src/cljs/athens/views/blocks.cljs

```diff
@@ -249,23 +249,49 @@
 
 ;; Helpers
 
-(defn on-change
-  [value uid]
-  ;; (prn "ONCHANGE" value)
-  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]])
-  ;; TODO: delete pages that are no longer connected to anything else
+
+(defn walk-parse-tree-for-links
+  [source-str link-fn db-fn]
   (parse/transform {:page-link (fn [& title]
                                  (let [inner-title (apply + title)]
                                    ;; `apply +` can return 0 if `title` is nil or empty string
                                    (when (and (string? inner-title)
-                                              (nil? (db/search-exact-node-title inner-title)))
+                                              (link-fn inner-title))
                                      (let [now (now-ts)
                                            uid (gen-block-uid)]
-                                       (dispatch [:transact [{:node/title  inner-title
-                                                              :block/uid   uid
-                                                              :edit/time   now
-                                                              :create/time now}]])))
-                                   (str "[[" inner-title "]]")))} (parser/parse-to-ast value)))
+                                       (db-fn inner-title now uid)))
+                                   (str "[[" inner-title "]]")))
+                    :hashtag   (fn [title]
```

Comment:
Does this also include `#[[]]` and `#[[nested [[links]]]]`?

Reply:
I think it will work because the parsing is bottom-up, but didn't know exactly. I'll test it

Reply:
Tested - it works with `#[[]]` but not with `#[[nested [[links]]]]`. Upon further inspection, the latter is a parser issue. Instead of fixing it here, I think it might be better to start a separate PR where I fix this issue and add some unit tests to prevent these problems from happening in the future.

Reply:
sounds good, thanks @HaojiXu